### PR TITLE
fix: consistent workspace status b/w CLI and UI

### LINKED ISSUE #2717: CLI displays workspace status as Running when Start was canceled and it was not started
CLI displays workspace status as Running when Start was canceled and it is not started

coder version:
Coder v0.7.6+a494489 Mon Jun 27 20:31:18 UTC 2022

Steps tp reproduce:
1. Create a workspace
2. Stop workspace
3. Start workspace and click Cancel 
4. in cli run `coder list`

Actual:
Status on UI: Canceled
Status in CLI: Running
Actual status - not running (stopped)

image

### PR DESCRIPTION
**Pull request by AbhineetJain**
This PR uses the same logic to show workspace status in CLI as the one used in UI.

## Subtasks

- [x] add a util method to compute workspace status and use it 
- [x] add unit tests 

Fixes #2717

### CODE REVIEW

File: cli/workspacestatus_internal_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			if got := workspaceStatus(tt.args.jobStatus, tt.args.transition); got != tt.want {
+				t.Errorf("workspaceStatus() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
```
Comment (johnstcn) [inline cli/workspacestatus_internal_test.go:118]: praise: nice test!

File: cli/workspacestatus.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	case codersdk.ProvisionerJobCanceling:
+		return "Canceling action"
+	case codersdk.ProvisionerJobCanceled:
+		return "Canceled action"
+	case codersdk.ProvisionerJobFailed:
+		return "Failed"
+	default:
+		return "Loading..."
+	}
+}
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply cli/workspacestatus.go:39]: Maybe, I'm not entirely sure yet where we'd use it. But it seems like a useful thing to have available!

[the remark this answers is not in the document]
Comment (AbhineetJain) [reply cli/workspacestatus.go:39]: We compute the workspace status in the exact same manner in frontend and display it on the workspace page. This ticket makes sure that we display the same status in CLI as the UI. Here's the UI code for example,

https://github.com/coder/coder/blob/main/site/src/util/workspace.ts#L22-L54

[the remark this answers is not in the document]
Comment (johnstcn) [reply cli/workspacestatus.go:39]: Ah yeah, I could imagine us returning the 'humanized' workspace state via the API as a field in that case, which would let us delete this code.
Probably better as a separate PR though!

**Review by johnstcn — APPROVED** [review]

File: cli/workspacestatus.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+var inProgressToStatus = map[codersdk.WorkspaceTransition]string{
+	codersdk.WorkspaceTransitionStart:  "Starting",
+	codersdk.WorkspaceTransitionStop:   "Stopping",
+	codersdk.WorkspaceTransitionDelete: "Deleting",
+}
+var succeededToStatus = map[codersdk.WorkspaceTransition]string{
+	codersdk.WorkspaceTransitionStart:  "Started",
+	codersdk.WorkspaceTransitionStop:   "Stopped",
+	codersdk.WorkspaceTransitionDelete: "Deleted",
+}
```
Comment (Emyrk) [inline cli/workspacestatus.go:16]: Maybe comment these? The naming of these was a bit confusing to me at first.

File: cli/workspacestatus.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (AbhineetJain) [reply cli/workspacestatus.go]: Moved to a separate file under the `codersdk` package.

[the remark this answers is not in the document]
Comment (AbhineetJain) [reply cli/workspacestatus.go]: Created an issue for removing duplicated logic: https://github.com/coder/coder/issues/2772

**Review by johnstcn — APPROVED** [review]

File: codersdk/workspacedisplaystatus.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+// Maps workspace transition to display status for Succeeded job status
+var succeededStatusFromTransition = map[WorkspaceTransition]string{
+	WorkspaceTransitionStart:  "Started",
+	WorkspaceTransitionStop:   "Stopped",
+	WorkspaceTransitionDelete: "Deleted",
+}
+
+const unknownStatus = "Unknown"
```
Comment (Emyrk) [inline codersdk/workspacedisplaystatus.go:17]: Should we make these enums? Idc if not, just asking.

**Review by Emyrk — APPROVED** [review]

File: codersdk/workspacedisplaystatus.go (hunk 4 above)
Comment (AbhineetJain) [reply codersdk/workspacedisplaystatus.go:17]: I was thinking about it tbh. Let's do it. All status strings, right?

Comment (AbhineetJain) [reply codersdk/workspacedisplaystatus.go:17]: Discussed via DM, and decided to defer this.

**Pull request merged**